File: app/src/main/java/com/sliebald/cula/ui/main/MainViewModel.java
package com.sliebald.cula.ui.main;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.sliebald.cula.data.database.Entities.LanguageEntry;
import com.sliebald.cula.utilities.InjectorUtils;


public class MainViewModel extends ViewModel {

    private final LiveData<LanguageEntry> mActiveLanguage;

    /**
     * Constructor of the ViewModel.
     */
    public MainViewModel() {
        mActiveLanguage = InjectorUtils.provideRepository().getActiveLanguage();
    }

    public LiveData<LanguageEntry> getActiveLanguage() {
        return mActiveLanguage;
    }

}

File: app/src/main/java/com/sliebald/cula/MainViewModel.kt
package com.sliebald.cula

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.sliebald.cula.data.database.entities.LanguageEntry
import com.sliebald.cula.utilities.InjectorUtils

class MainViewModel : ViewModel() {
    val activeLanguage: LiveData<LanguageEntry?> = InjectorUtils.provideRepository().getActiveLanguage()
}